### PR TITLE
Add CRUD to the home controller

### PR DESCRIPTION
**Pull request by ogbeidebest**
Add Create, edit, and delete functionality to the home controller for the user. 

Linear ticket: https://linear.app/team-socket/issue/BAC-68/create-edit-and-delete-review

### CODE REVIEW

File: src/Controllers/HomeController.cs [hunk 1]
```diff
@@ -59,16 +59,33 @@ public ActionResult<IEnumerable<ReviewForDisplayDto>> PostedReviews()
         /// <param name="CreateReview"></param>
         /// <returns></returns>
         [SwaggerOperation(Summary = "Create a Review with this endpoint")]
-        [HttpGet("create")]
-        public ActionResult CreateReview([FromBody] ReviewForCreationDto reviewForCreationDto)
+        [HttpPost("create")]
```
Comment (DCWhiteSnake) [inline src/Controllers/HomeController.cs]: change this to review/. So it's a POST /api/review to create a review.

File: src/Controllers/HomeController.cs [hunk 2]
```diff
@@ -102,18 +119,21 @@ public IActionResult Postreview(Review review)
...
-        /// <returns>Ok</returns>
-        [SwaggerOperation(Summary = "Deletes All Reviews Associated With this user")]
-        [HttpDelete("reviews/delete-all-reviews")]
-        public IActionResult DeleteReviews()
+        /// <param name="reviewId"></param>
+        /// <returns>Review is successfully deleted</returns>
+        [SwaggerOperation(Summary = "delete a review by an User")]
+        [HttpDelete]
+        [Authorize(Roles = "Customer", AuthenticationSchemes = "Bearer")]
+        [Route("review/{reviewId}/delete")]
```
Comment (DCWhiteSnake) [inline src/Controllers/HomeController.cs]: should be Delete review/reviewId

**Comment by DCWhiteSnake** [comment]
Were all these additions tested in the swagger UI?

File: src/Controllers/HomeController.cs (hunk 2 above)
Comment (ogbeidebest) [reply src/Controllers/HomeController.cs]: Done

File: src/Controllers/HomeController.cs (hunk 1 above)
Comment (ogbeidebest) [reply src/Controllers/HomeController.cs]: Done

File: src/Controllers/HomeController.cs [hunk 3]
```diff
@@ -59,16 +61,38 @@ public ActionResult<IEnumerable<ReviewForDisplayDto>> PostedReviews()
...
+           
+        }
+
+        [SwaggerOperation(Summary = "Update a review by an user")]
+        [HttpPut]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        [SwaggerResponseExample(400, typeof(BadCustomerAccountForCreationExample))]
+        [SwaggerResponseExample(200, typeof(GoodCustomerAccountForCreationExample))]
```
Comment (DCWhiteSnake) [inline src/Controllers/HomeController.cs:79]: Why are you using account creation example for review editing example

File: src/Controllers/HomeController.cs [hunk 4]
```diff
@@ -102,18 +126,22 @@ public IActionResult Postreview(Review review)
...
-        [SwaggerOperation(Summary = "Deletes All Reviews Associated With this user")]
-        [HttpDelete("reviews/delete-all-reviews")]
-        public IActionResult DeleteReviews()
+        /// <param name="reviewId"></param>
+        /// <returns>Review is successfully deleted</returns>
+        [SwaggerOperation(Summary = "delete a review by an User")]
+        [HttpDelete]
+        [Authorize(Roles = "Customer", AuthenticationSchemes = "Bearer")]
+        [Route("review/{reviewId}")]
+        public IActionResult DeleteReviews(Guid reviewId)
```
Comment (DCWhiteSnake) [inline src/Controllers/HomeController.cs:135]: Delete reviews but you're deleting  a single review? You need a separate action for deleting all reviews and deleting a single review

Comment (ogbeidebest) [reply src/Controllers/HomeController.cs:135]: I just edited the method i was there before. Let me close the pr and create it again

**Pull request closed without merging**